docs/i2c.rst

```diff
@@ -21,6 +21,17 @@ should be described separately in that device's documentation.
 Functions
 =========
 
+.. py:function:: init(freq=None, sda=None, scl=None)
+
+    Re-initialize peripheral with the specified frequency ``freq`` on the
+    specified ``sda`` and ``scl`` pins. If pins are not specified, ``pin20``
+    and ``pin19`` are used. If frequency is not specified, it's not changed.
+
+    .. warning::
+
+        Changing the I²C pins from defaults will make the accelerometer and
+        compass stop working, as they are connected internally to those pins.
```

Comment:
There are actually 2 peripherals which can act in either I2C or SPI mode.  One of those periphs are configured as I2C for compass and accelerometer.  The other is uninitialised and will be used in SPI mode if the user ever calls spi.init().

So, while it's possible to have 2 I2C periphs (eg 1 for accel/compass, another for the user) I think it's simpler to have one peripheral reserved for I2C and the other for SPI.  This maps to the current microbit module where there's 1 i2c object and 1 spi object.

The alternative is to allow the user to specify which periph they want and in what mode and with what pins.  But that will really complicate the API.